lib/Subscribers/SubscriberListingRepository.php

```diff
@@ -130,10 +130,17 @@ protected function applyFilters(QueryBuilder $queryBuilder, array $filters) {
       return;
     }
     if ($filters['segment'] === self::FILTER_WITHOUT_LIST) {
-      $queryBuilder->leftJoin('s.subscriberSegments', 'ssg', Join::WITH, (string)$queryBuilder->expr()->eq('ssg.status', ':statusSubscribed'))
+      $qbSubselect = $this->entityManager->createQueryBuilder();
+      $qbSubselect->select('s2.id')
+        ->from(SubscriberEntity::class, 's2')
+        ->leftJoin('s2.subscriberSegments', 'ssg2', Join::WITH, (string)$qbSubselect->expr()->eq('ssg2.status', ':statusSubscribed'))
+        ->join('ssg2.segment', 'sg2', Join::WITH, (string)$qbSubselect->expr()->isNull('sg2.deletedAt'));
+
+        $queryBuilder->leftJoin('s.subscriberSegments', 'ssg', Join::WITH, (string)$queryBuilder->expr()->eq('ssg.status', ':statusSubscribed'))
         ->leftJoin('ssg.segment', 'sg', Join::WITH, (string)$queryBuilder->expr()->isNull('sg.deletedAt'))
         ->andWhere('s.deletedAt IS NULL')
         ->andWhere('sg.id IS NULL')
+        ->andWhere($queryBuilder->expr()->notIn('s.id', $qbSubselect->getDQL()))
```

Comment:
I think this might cause performance issues. As far as I know, subqueries in WHERE IN are executed for each row.
I see some other conditions so planner should probably use those to limit rows in result first, but still, it might be an issue for users who have more users with no list. And also it might be safer not to rely too much on MySQL planner.

I think we could either execute the subselect and use the result to avoid it or I think there is some trick using the subquery within JOIN. I think that in this case we could probably add left join and join the subquery and in where clause we could then use something like `s2.id IS NULL` 🤔